client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -50,46 +56,52 @@ public ReaderGroupManagerImpl(String scope, Controller controller, ClientFactory
         return new StreamImpl(scope, streamName);
     }
     
     @Override
     public ReaderGroup createReaderGroup(String groupName, ReaderGroupConfig config, Set<String> streams) {
         NameUtils.validateReaderGroupName(groupName);
-        createStreamHelper(NameUtils.getStreamForReaderGroup(groupName),
-                           StreamConfiguration.builder()
-                                              .scope(scope)
-                                              .streamName(NameUtils.getStreamForReaderGroup(groupName))
-                                              .scalingPolicy(ScalingPolicy.fixed(1))
-                                              .build());
+        createStreamHelper(getStreamForReaderGroup(groupName), StreamConfiguration.builder()
+                                                                                  .scope(scope)
+                                                                                  .streamName(getStreamForReaderGroup(groupName))
+                                                                                  .scalingPolicy(ScalingPolicy.fixed(1))
+                                                                                  .build());
         SynchronizerConfig synchronizerConfig = SynchronizerConfig.builder().build();
-        ReaderGroupImpl result = new ReaderGroupImpl(scope,
-                                                     groupName,
-                                                     synchronizerConfig,
-                                                     new JavaSerializer<>(),
-                                                     new JavaSerializer<>(),
-                                                     clientFactory,
-                                                     controller);
+        ReaderGroupImpl result = new ReaderGroupImpl(scope, groupName, synchronizerConfig, new JavaSerializer<>(),
+                                                     new JavaSerializer<>(), clientFactory, controller);
         result.initializeGroup(config, streams);
         return result;
     }
-    
+
+    @Override
+    public void deleteReaderGroup(String groupName) {
+        getAndHandleExceptions(controller.sealStream(scope, getStreamForReaderGroup(groupName))
+                                         .thenCompose(b -> controller.deleteStream(scope,
+                                                                                   getStreamForReaderGroup(groupName)))
+                                         .exceptionally(e -> {
+                                             if (e instanceof InvalidStreamException) {
+                                                 return null;
+                                             } else {
+                                                 log.warn("Failed to delete stream", e);
+                                             }
+                                             throw Lombok.sneakyThrow(e);
```

Comment:
That's true but the compiler does not know it. E is of type Throwable here, because that is the contract on the exceptionally method. So this line is equivalent to `throw e;` but that would be a compile error.